src/platforms/esp32/main/platform_nifs.c

```diff
@@ -0,0 +1,187 @@
+/***************************************************************************
+ *   Copyright 2019 by Fred Dushin <fred@dushin.net>                       *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU Lesser General Public License as        *
+ *   published by the Free Software Foundation; either version 2 of the    *
+ *   License, or (at your option) any later version.                       *
+ *                                                                         *
+ *   This program is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU General Public License     *
+ *   along with this program; if not, write to the                         *
+ *   Free Software Foundation, Inc.,                                       *
+ *   51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA .        *
+ ***************************************************************************/
+
+#define _GNU_SOURCE
+
+#include "defaultatoms.h"
+#include "nifs.h"
+#include "memory.h"
+#include "term.h"
+
+#include <esp_system.h>
+
+//#define ENABLE_TRACE
+#include "trace.h"
+
+#define VALIDATE_VALUE(value, verify_function) \
+    if (UNLIKELY(!verify_function((value)))) { \
+        argv[0] = ERROR_ATOM; \
+        argv[1] = BADARG_ATOM; \
+        return term_invalid_term(); \
+    }
+
+#define RAISE_ERROR(error_type_atom) \
+    ctx->x[0] = ERROR_ATOM; \
+    ctx->x[1] = (error_type_atom); \
+    return term_invalid_term();
+
+static const char *const esp_rst_unknown_atom   = "\xF"  "esp_rst_unknown";
+static const char *const esp_rst_poweron        = "\xF"  "esp_rst_poweron";
+static const char *const esp_rst_ext            = "\xB"  "esp_rst_ext";
+static const char *const esp_rst_sw             = "\xA"  "esp_rst_sw";
+static const char *const esp_rst_panic          = "\xD"  "esp_rst_panic";
+static const char *const esp_rst_int_wdt        = "\xF"  "esp_rst_int_wdt";
+static const char *const esp_rst_task_wdt       = "\x10" "esp_rst_task_wdt";
+static const char *const esp_rst_wdt            = "\xB"  "esp_rst_wdt";
+static const char *const esp_rst_deepsleep      = "\x11" "esp_rst_deepsleep";
+static const char *const esp_rst_brownout       = "\x10" "esp_rst_brownout";
+static const char *const esp_rst_sdio           = "\xC"  "esp_rst_sdio";
+//                                                        123456789ABCDEF01
+
+//
+// NIFs
+//
+    
+static term nif_esp_random(Context *ctx, int argc, term argv[])
+{
+    if (argc != 0) {
```

Comment:
Function signature strings already have arity, so this function cannot be called with a wrong number of arguments by mistake.
So we can safely remove this check.